File: binary-tree-level-order-traversal/binary-tree-level-order-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        Queue<TreeNode> que = new LinkedList<>();
        List<List<Integer>> result = new ArrayList<>();
        if(root == null){
            return result;
        }
        
        que.offer(root);
        
        while(!que.isEmpty()){
            int n = que.size();
            List<Integer> levelList = new ArrayList<>();
            for(int i = 0; i< n; i++){
                TreeNode current = que.poll();
                levelList.add(current.val);
                if(current.left != null){
                    que.add(current.left);
                }
                if(current.right != null){
                    que.add(current.right);
                }
            }
            result.add(levelList);
        }
        return result;
        
    }
}

File: binary-tree-level-order-traversal/binary-tree-level-order-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        que = []
        result  = []
        if not root:
            return result
        
        que.append(root)
        
        while que:
            tmp = []
            n = len(que)
            for i in range(n):
                node = que.pop(0)
                tmp.append(node.val)
                if node.left:
                    que.append(node.left)
                if node.right:
                    que.append(node.right)
            result.append(tmp)
        return result
        
                
        